### PR TITLE
removed `pallet::getter` from `pallet-sudo`

### PR DESCRIPTION
**Pull request by muraca**
part of #3326 

@ggwpez @kianenigma @shawntabrizi

### CODE REVIEW

**Review by ggwpez — APPROVED** [review]

File: substrate/frame/sudo/src/extension.rs [hunk 1]
```diff
@@ -86,7 +86,7 @@ where
 		info: &DispatchInfoOf<Self::Call>,
 		_len: usize,
 	) -> TransactionValidity {
-		let sudo_key: T::AccountId = <Pallet<T>>::key().ok_or(UnknownTransaction::CannotLookup)?;
+		let sudo_key: T::AccountId = <Key<T>>::get().ok_or(UnknownTransaction::CannotLookup)?;
```
Comment (gupnik) [inline substrate/frame/sudo/src/extension.rs:89]: I will do `Key::<T>::get()` here as well.

**Review by gupnik — APPROVED** [review]

File: substrate/frame/sudo/src/extension.rs (hunk 1 above)
Comment (kianenigma) [reply substrate/frame/sudo/src/extension.rs:89]: Yes please 🙏🏻

Comment (kianenigma) [inline substrate/frame/sudo/src/extension.rs:89]:
```suggestion
		let sudo_key: T::AccountId = Key::<T>::get().ok_or(UnknownTransaction::CannotLookup)?;
```

**Comment by muraca** [comment]
@ggwpez the silent label was removed as I had to write a prdoc because these are breaking changes

**Comment by ggwpez** [comment]
How is this a breaking change?

**Comment by muraca** [comment]
All getters visibility is `pub`, while the storage items left behind might not have this visibility, for example here we have `pub(super)`. 
This is probably due to the fact that the key is managed by pallet logic and we don't want other runtime logic to modify it. 

`pallet_sudo::Key::<T>::get()` is not accessible from other pallets.

**Comment by kianenigma** [comment]
Yeah, so I also first added `Silent` to these and added to the queue, but then removed them. 

If we are to be pedantic, they are all breaking changes.

**Comment by muraca** [comment]
> `pallet_sudo::Key::<T>::get()` is not accessible from other pallets.

what should we do with this issue?

**Comment by kianenigma** [comment]
One solution would be to just remove the usage of all getters in pallets one by one, but keep the getters. 

This will not be breaking change in any way, and do 99% of the work.

Then, in one breaking PR, we mark the getters as deprecated and/or remove them.

**Comment by muraca** [comment]
> One solution would be to just remove the usage of all getters in pallets one by one, but keep the getters.

Keeping the getter would not solve the issue, if you want to not use it.
Say that a pallet is using the sudo key, for whatever reason. If not with a getter, how will this other pallet access such data?   

The solutions I see are either `pub` for all the storage items where we remove a getter, or we manually implement getters for items that must not be changed by other logic, for example the sudo key.

**Pull request merged**